locations/spiders/bc_pizza.py

```diff
@@ -1,95 +1,136 @@
 import re
-
 import scrapy
-
+from datetime import datetime
+from locations.hours import OpeningHours
 from locations.items import GeojsonPointItem
+from locations.google_url import extract_google_position
 
-regex_am = r"\s?([Aa][Mm])"
-regex_pm = r"\s?([Pp][Mm])"
-regex_hours = (
-    r"\d{1,2}:\d{1,2}\s?[Aa]?[Pp]?[Mm]\s?-\s?\d{1,2}:\d{1,2}\s?" r"[Aa]?[Pp]?[Mm]"
-)
+# source: https://gist.github.com/rogerallen/1583593
+us_state_to_abbrev = {
+    "Alabama": "AL",
+    "Alaska": "AK",
+    "Arizona": "AZ",
+    "Arkansas": "AR",
+    "California": "CA",
+    "Colorado": "CO",
+    "Connecticut": "CT",
+    "Delaware": "DE",
+    "Florida": "FL",
+    "Georgia": "GA",
+    "Hawaii": "HI",
+    "Idaho": "ID",
+    "Illinois": "IL",
+    "Indiana": "IN",
+    "Iowa": "IA",
+    "Kansas": "KS",
+    "Kentucky": "KY",
+    "Louisiana": "LA",
+    "Maine": "ME",
+    "Maryland": "MD",
+    "Massachusetts": "MA",
+    "Michigan": "MI",
+    "Minnesota": "MN",
+    "Mississippi": "MS",
+    "Missouri": "MO",
+    "Montana": "MT",
+    "Nebraska": "NE",
+    "Nevada": "NV",
+    "New Hampshire": "NH",
+    "New Jersey": "NJ",
+    "New Mexico": "NM",
+    "New York": "NY",
+    "North Carolina": "NC",
+    "North Dakota": "ND",
+    "Ohio": "OH",
+    "Oklahoma": "OK",
+    "Oregon": "OR",
+    "Pennsylvania": "PA",
+    "Rhode Island": "RI",
+    "South Carolina": "SC",
+    "South Dakota": "SD",
+    "Tennessee": "TN",
+    "Texas": "TX",
+    "Utah": "UT",
+    "Vermont": "VT",
+    "Virginia": "VA",
+    "Washington": "WA",
+    "West Virginia": "WV",
+    "Wisconsin": "WI",
+    "Wyoming": "WY",
+    "District of Columbia": "DC",
+    "American Samoa": "AS",
+    "Guam": "GU",
+    "Northern Mariana Islands": "MP",
+    "Puerto Rico": "PR",
+    "United States Minor Outlying Islands": "UM",
+    "U.S. Virgin Islands": "VI",
+}
```

Comment:
Semi recently I tried to remove these state arrays from spiders, swapping most of them to `GeonamesCache().get_us_states()`. In this case, I don't think it's that important to go from state name to state code, but if it was, it should be done in pipeline code so all US spiders benefit, not just this one.